### PR TITLE
Display fn spec in the doc buffers

### PR DESCRIPTION
**Pull request by cskksc**
We get the formatted spec strings through the middleware `:info` op, and we just display those in the doc buffers. Depends on clojure-emacs/cider-nrepl#374

- [ ] The commits are consistent with our [contribution guidelines][1]
- [ ] You've added tests (if possible) to cover your change(s)
- [x] All tests are passing (`make test`)
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [x] You've updated the changelog (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)
- [ ] You've updated the refcard (if you made changes to the commands listed there)

Thanks!

[1]: https://github.com/clojure-emacs/cider/blob/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: cider-doc.el [hunk 1]
```diff
@@ -455,6 +456,10 @@ Tables are marked to be ignored by line wrap."
                                         (browse-url (button-get x 'url))))
           (insert ".\n"))
         (insert "\n")
+        (when spec
+          (emit "Spec: " 'font-lock-function-name-face)
+          (mapc (lambda (s) (insert s "\n")) spec))
+        (insert "\n")
```
Comment (Malabarba) [inline cider-doc.el]: I think this line should be inside the `(when ...)

**Comment by Malabarba** [comment]
:+1: besides the comment

**Comment by bbatsov** [comment]
Looks good to me as well. 

On an unrelated note - probably it would make sense to add some command to display just the spec in some fancier manner (e.g. inline). Enriching eldoc with spec info would also be pretty cool.

File: cider-doc.el (hunk 1 above)
Comment (cskksc) [inline cider-doc.el]: Done :-)

**Comment by cskksc** [comment]
> Enriching eldoc with spec info would also be pretty cool.

Yep, it would be. But I think for most users, `eldoc-echo-area-use-multiline-p` would be set to `truncate-sym-name-if-fit`, the default value. So the multiline strings won't be displayed. Should we still add it in ? It would be displayed when `eldoc-echo-area-use-multiline-p` is t. 

> probably it would make sense to add some command to display just the spec in some fancier manner (e.g. inline)

I'm not sure what `inline` is. Do we have some other feature that uses this ? I could figure it out from there.

**Comment by bbatsov** [comment]
> Yep, it would be. But I think for most users, eldoc-echo-area-use-multiline-p would be set to truncate-sym-name-if-fit, the default value. So the multiline strings won't be displayed. Should we still add it in ? It would be displayed when eldoc-echo-area-use-multiline-p is t.

Well, that's why we have a manual. Curious users will figure this out. :-)

> I'm not sure what inline is. Do we have some other feature that uses this ? I could figure it out from there :-)

I call the use of overlays "inline". Not a big deal, just something that crossed my mind.

**Comment by cskksc** [comment]
@bbatsov Oh, ok. Got it. We can do something like this. I'll post a separate PR for that. The eldoc integration is pretty straightforward. Lets include that here.

**Pull request merged**